File: PST/FUEL/src/FUEL/Simple3Des.vb
Imports Microsoft.VisualBasic.CompilerServices
Imports System
Imports System.IO
Imports System.Security.Cryptography
Imports System.Text

Namespace FUEL
    Friend NotInheritable Class Simple3Des
        ' Methods
        Public Sub New(ByVal key As String)
            Me.TripleDes.Key = Me.TruncateHash(key, (Me.TripleDes.KeySize / 8))
            Me.TripleDes.IV = Me.TruncateHash("", (Me.TripleDes.BlockSize / 8))
        End Sub

        Public Function DecryptData(ByVal encryptedtext As String) As String
            Dim buffer As Byte() = Convert.FromBase64String(encryptedtext)
            Dim stream2 As New MemoryStream
            Dim stream As New CryptoStream(stream2, Me.TripleDes.CreateDecryptor, CryptoStreamMode.Write)
            stream.Write(buffer, 0, buffer.Length)
            stream.FlushFinalBlock
            Return Encoding.Unicode.GetString(stream2.ToArray)
        End Function

        Public Function EncryptData(ByVal PlainText As String) As String
            Dim bytes As Byte() = Encoding.Unicode.GetBytes(PlainText)
            Dim stream2 As New MemoryStream
            Dim stream As New CryptoStream(stream2, Me.TripleDes.CreateEncryptor, CryptoStreamMode.Write)
            stream.Write(bytes, 0, bytes.Length)
            stream.FlushFinalBlock
            Return Convert.ToBase64String(stream2.ToArray)
        End Function

        Private Function TruncateHash(ByVal Key As String, ByVal Length As Integer) As Byte()
            Return DirectCast(Utils.CopyArray(DirectCast(New SHA256CryptoServiceProvider().ComputeHash(Encoding.Unicode.GetBytes(Key)), Array), New Byte(((Length - 1) + 1)  - 1) {}), Byte())
        End Function


        ' Fields
        Private TripleDes As TripleDESCryptoServiceProvider = New TripleDESCryptoServiceProvider
    End Class
End Namespace

File: PST/FUEL Source Code/FUEL/Backup/Simple3Des.cs
// Decompiled with JetBrains decompiler
// Type: FUEL.Simple3Des
// Assembly: FUEL, Version=2.4.8.0, Culture=neutral, PublicKeyToken=null
// MVID: A3D630D0-0620-425D-8CFB-B67B2AC75507
// Assembly location: C:\hp\FUEL\FUEL.dll

using System;
using System.IO;
using System.Security.Cryptography;
using System.Text;

namespace FUEL
{
  internal sealed class Simple3Des
  {
    private TripleDESCryptoServiceProvider TripleDes;

    public Simple3Des(string key)
    {
      this.TripleDes = new TripleDESCryptoServiceProvider();
      this.TripleDes.Key = this.TruncateHash(key, this.TripleDes.KeySize / 8);
      this.TripleDes.IV = this.TruncateHash("", this.TripleDes.BlockSize / 8);
    }

    private byte[] TruncateHash(string Key, int Length) => (byte[]) Microsoft.VisualBasic.CompilerServices.Utils.CopyArray((Array) new SHA256CryptoServiceProvider().ComputeHash(Encoding.Unicode.GetBytes(Key)), (Array) new byte[checked (Length - 1 + 1)]);

    public string EncryptData(string PlainText)
    {
      byte[] bytes = Encoding.Unicode.GetBytes(PlainText);
      MemoryStream memoryStream = new MemoryStream();
      CryptoStream cryptoStream = new CryptoStream((Stream) memoryStream, this.TripleDes.CreateEncryptor(), CryptoStreamMode.Write);
      cryptoStream.Write(bytes, 0, bytes.Length);
      cryptoStream.FlushFinalBlock();
      return Convert.ToBase64String(memoryStream.ToArray());
    }

    public string DecryptData(string encryptedtext)
    {
      byte[] buffer = Convert.FromBase64String(encryptedtext);
      MemoryStream memoryStream = new MemoryStream();
      CryptoStream cryptoStream = new CryptoStream((Stream) memoryStream, this.TripleDes.CreateDecryptor(), CryptoStreamMode.Write);
      cryptoStream.Write(buffer, 0, buffer.Length);
      cryptoStream.FlushFinalBlock();
      return Encoding.Unicode.GetString(memoryStream.ToArray());
    }
  }
}
